helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -695,12 +697,6 @@ private void handleEvent(ClusterEvent event, BaseControllerDataProvider dataProv
     event.addAttribute(AttributeName.STATEFUL_REBALANCER.name(),
         _rebalancerRef.getRebalancer(manager));
 
-    if (!manager.isLeader()) {
```

Comment:
Delaying this check might cause confusing and seems not necessary to this PR. Let's don't change it in this PR even if desired.

Reply:
Ah good catch. I intended to replace this line with the new one: check leader and get session at the same time. But somehow it goes to later right before invoking pipeline. Resolved.